syntax = "proto3";

option go_package = "github.com/mingkaic/tenncor/lead";

package tenncor;

import "google/protobuf/any.proto";

message DoubleArr { repeated double data = 1; }
message FloatArr { repeated float data = 1; }
message ByteArr { bytes data = 1; }
message Int32Arr { repeated int32 data = 1; }
message Int64Arr { repeated int64 data = 1; }
message Uint32Arr { repeated uint32 data = 1; }
message Uint64Arr { repeated uint64 data = 1; }

message TensorPb
{
	// takes messages DoubleArr, ...,
	// Uint64Arr depending on dtype
	google.protobuf.Any data = 1;
	repeated uint64 shape = 2;
	uint32 type = 3;
}

message DataRepoPb
{
	string gid = 1;
	map<string, TensorPb> data_map = 2;
}
